tempesta_fw/http_parser.c

```diff
@@ -103,17 +92,105 @@ memchreol(const unsigned char *s, size_t n)
 #define IS_WS(c) (c == ' ' || c == '\t')
 
 /**
+ * When parsing it's necessary to not forget to update @data and @len
+ * properly. So, presence of __data_{} macroses will be helpful to reduce amount
+ * of direct @data/@len manipulations.
+ */
+#define __data_offset(pos)						\
+	(size_t)((pos) - data)
+#define __data_margin(pos)						\
```

Comment:
I believe this is supposed to mean the number of bytes that is not processed yet. `margin` is a very confusing term for that purpose. I would suggest to use the term `remainder`, like `__data_remainder()`